airflow/providers/http/hooks/http.py

```diff
@@ -126,15 +126,20 @@ def run(
         else:
             url = (self.base_url or '') + (endpoint or '')
 
+        request_parameters = dict(method=self.method.upper(), url=url, headers=headers)
+        request_parameters.update(request_kwargs)
+
         if self.method == 'GET':
             # GET uses params
-            req = requests.Request(self.method, url, params=data, headers=headers, **request_kwargs)
+            request_parameters["params"] = data
         elif self.method == 'HEAD':
             # HEAD doesn't use params
-            req = requests.Request(self.method, url, headers=headers, **request_kwargs)
+            pass
         else:
             # Others use data
-            req = requests.Request(self.method, url, data=data, headers=headers, **request_kwargs)
+            request_parameters["data"] = data
+
+        req = requests.Request(**request_parameters)
```

Comment:
This block appears to change the code to have no difference in behaviour, unless I've missed something

Can you remove hunk please?

Reply:
Thanks, that was garbage after latest changes. Removed.